src/main/java/org/apache/commons/compress/archivers/tar/TarArchiveEntry.java

```diff
@@ -368,10 +397,74 @@ public TarArchiveEntry(final File file) {
      * @param fileName the name to be used for the entry.
      */
     public TarArchiveEntry(final File file, final String fileName) {
+        final String normalizedName = normalizeFileName(fileName, false);
+        this.file = file.toPath();
+
+        try {
+            readFileMode(this.file, normalizedName);
+        } catch (IOException e) {
+            // Ignore for backwards compatibility
```

Comment:
Similar things here.
Hope we could find a way to handle the exception here. Currently I got no good ideas for this.

Reply:
Maybe we could just rethrow the exception without declearing in the constructor?

Reply:
I don't think this will ever work without ignoring the exception here. Have a look at TarArchiveEntryTest#testTarFileWithFSRoot. On Windows the entries are directly added from the C:\ drive. They don't have to exist that the test passes with the old File API. But the NIO API tries to read the file size from disk. When it is not present it will throw an NoSuchFileException.

Also throwing a runtime exception here is in my opinion a really bad idea. This is totally unexpected for the user since he only notices it at runtime. With a checked exception he would directly notice it after upgrading. But since changing the API is no option I think we are only left with swallowing the error.

Reply:
Ignoring exceptions here - and documenting that we do - seems to be the best approach.

Another alternative would be to not change this constructor's code to use the new methods at all. If we want to invoke the new methods then I'd suggest to set `modTime` and  `size` "the old way" before invoking the new methods so we keep the old behaviour if things go wrong in NIO land.

Reply:
> Ignoring exceptions here - and documenting that we do - seems to be the best approach.

Seems it's the best what we can do here.

Reply:
> Ignoring exceptions here - and documenting that we do - seems to be the best approach.

Added a note to the javadoc and mentioned that the exceptions can be handled when using the Path constructor.

> Another alternative would be to not change this constructor's code to use the new methods at all.

This would already create a feature gap between the file and the path constructors. The Path constructor can add the uid and gid of the file on linux. So this is not really preferred by me.



> If we want to invoke the new methods then I'd suggest to set `modTime` and `size` "the old way" before invoking the new methods so we keep the old behaviour if things go wrong in NIO land.

I implemented your suggestion with a litte change. `modTime` and `size` will be only added "the old way" when calling the NIO methods throws an exception. Hope this is ok.